client/securedrop_client/gui/source/delete/dialog.py

```diff
@@ -26,23 +26,36 @@
 class DeleteSourceDialog(ModalDialog):
     """Used to confirm deletion of source accounts."""
 
-    def __init__(self, source: Source) -> None:
+    def __init__(self, sources: set[Source]) -> None:
         super().__init__(show_header=False, dangerous=True)
+        self.sources = sources
 
-        self.source = source
+        # If the dialog is constructed with no sources, show a warning; otherwise,
+        # confirm the number and designation of the sources to be deleted
+        count = len(sources)
+        if count == 0:
+            self._show_warning_nothing_selected()
+        else:
+            if count == 1:
```

Comment:
This should use ngettext instead of manually implementing the plural.

Reply:
Thank you , resolved in https://github.com/freedomofpress/securedrop-client/pull/2188/commits/16e48a80d1d74758c5503d4c48c851aa5db5fb63